clang/include/clang/Sema/Sema.h

```diff
@@ -12981,7 +12981,8 @@ static bool handleMaxWorkSizeAttrExpr(Sema &S, const AttrInfo &AI,
     if (ArgVal->isNegative()) {
       S.Diag(Expr->getExprLoc(),
              diag::warn_attribute_requires_non_negative_integer_argument)
-          << &AI << Idx << Expr->getSourceRange();
+          << S.Context.IntTy << S.Context.UnsignedIntTy
```

Comment:
I think you should get the type out of the expression rather than hard-code `IntTy` -- the user could have done something like used an integer suffix that would give a different type, and the diagnostic would be somewhat confusing. e.g., `0LL` instead of `0`.

You should use `UnsignedLongLongTy` for the unsigned type as that's what we always wind up converting it to anyway.

Reply:
Thanks @AaronBallman for lookin into the patch.
If i understand correctly, do you mean to use something like << S.Context.IntTy << S.Context.UnsignedLongLongTy

Reply:
Nope, sorry for not being more clear. I mean something more like `<< Expr->getType() << S.Context.UnsignedLongLongTy`

(as a drive-by, we may want to rename the variable `Expr` in this function to something that doesn't conflict with the type named `Expr`.)

Reply:
Thanks for the explanation. Yes, it makes sense to me to use UnsignedLongLongTy for the unsigned type. I have renamed he variable Expr to avoid the conflicts with type names Expr.